src/Type/Php/CtypeDigitFunctionTypeSpecifyingExtension.php

```diff
@@ -59,9 +61,17 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 		$unionType = TypeCombinator::union(...$types);
 		$specifiedTypes = $this->typeSpecifier->create($exprArg, $unionType, $context, false, $scope);
 
-		if ($context->true() && $exprArg instanceof Cast) {
+		if ($context->true() && $exprArg instanceof Cast\String_) {
```

Comment:
with this change, I just turn the narrowing for non-string casts back to what it was before the recent PR https://github.com/phpstan/phpstan-src/pull/2186